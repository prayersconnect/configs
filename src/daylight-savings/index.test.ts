import { getDSTEnd, getDSTStart } from './index';
import dstTransitionDates from './list';
import cases from 'jest-in-case';

const countries = Object.keys(dstTransitionDates);

function getRangeTestObject(year: number) {
  return (country: string) => {
    return {
      name: `${country} ${year}`,
      year,
      country,
    };
  };
}

describe('daylight savings', () => {
  describe('getDSTStart', () => {
    it('returns null for invalid country', () => {
      expect(getDSTStart('invalid', 2018)).toBeNull();
    });

    it('returns null for invalid year', () => {
      expect(getDSTStart('US', 2019)).toBeNull();
    });

    cases(
      'returns correct start date',
      (opts: { name: string; year: number; country: string }) => {
        expect(getDSTStart(opts.country, opts.year)).toMatchSnapshot();
      },
      countries
        .map(getRangeTestObject(2021))
        .concat(countries.map(getRangeTestObject(2022)))
        .concat(countries.map(getRangeTestObject(2023)))
        .concat(countries.map(getRangeTestObject(2024)))
    );

    cases(
      'returns correct end date',
      (opts: { name: string; year: number; country: string }) => {
        expect(getDSTEnd(opts.country, opts.year)).toMatchSnapshot();
      },
      countries
        .map(getRangeTestObject(2021))
        .concat(countries.map(getRangeTestObject(2022)))
        .concat(countries.map(getRangeTestObject(2023)))
        .concat(countries.map(getRangeTestObject(2024)))
    );
  });
});
