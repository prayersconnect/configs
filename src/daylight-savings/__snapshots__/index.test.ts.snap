// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`daylight savings getDSTEnd returns correct end date Australia 2021 1`] = `"2022-04-02"`;

exports[`daylight savings getDSTEnd returns correct end date Australia 2022 1`] = `"2023-04-01"`;

exports[`daylight savings getDSTEnd returns correct end date Australia 2023 1`] = `"2024-04-06"`;

exports[`daylight savings getDSTEnd returns correct end date Australia 2024 1`] = `"2025-04-05"`;

exports[`daylight savings getDSTEnd returns correct end date Canada 2021 1`] = `"2021-11-07"`;

exports[`daylight savings getDSTEnd returns correct end date Canada 2022 1`] = `"2022-11-05"`;

exports[`daylight savings getDSTEnd returns correct end date Canada 2023 1`] = `"2023-11-04"`;

exports[`daylight savings getDSTEnd returns correct end date Canada 2024 1`] = `"2024-11-02"`;

exports[`daylight savings getDSTEnd returns correct end date United Kingdom 2021 1`] = `"2021-10-30"`;

exports[`daylight savings getDSTEnd returns correct end date United Kingdom 2022 1`] = `"2022-10-29"`;

exports[`daylight savings getDSTEnd returns correct end date United Kingdom 2023 1`] = `"2023-10-28"`;

exports[`daylight savings getDSTEnd returns correct end date United Kingdom 2024 1`] = `"2024-10-26"`;

exports[`daylight savings getDSTEnd returns correct end date United States 2021 1`] = `"2021-11-06"`;

exports[`daylight savings getDSTEnd returns correct end date United States 2022 1`] = `"2022-11-05"`;

exports[`daylight savings getDSTEnd returns correct end date United States 2023 1`] = `"2023-11-04"`;

exports[`daylight savings getDSTEnd returns correct end date United States 2024 1`] = `"2024-11-02"`;

exports[`daylight savings getDSTStart returns correct start date Australia 2021 1`] = `"2021-10-03"`;

exports[`daylight savings getDSTStart returns correct start date Australia 2022 1`] = `"2022-10-02"`;

exports[`daylight savings getDSTStart returns correct start date Australia 2023 1`] = `"2023-10-01"`;

exports[`daylight savings getDSTStart returns correct start date Australia 2024 1`] = `"2024-10-07"`;

exports[`daylight savings getDSTStart returns correct start date Canada 2021 1`] = `"2021-03-14"`;

exports[`daylight savings getDSTStart returns correct start date Canada 2022 1`] = `"2022-03-13"`;

exports[`daylight savings getDSTStart returns correct start date Canada 2023 1`] = `"2023-03-12"`;

exports[`daylight savings getDSTStart returns correct start date Canada 2024 1`] = `"2024-03-10"`;

exports[`daylight savings getDSTStart returns correct start date United Kingdom 2021 1`] = `"2021-03-28"`;

exports[`daylight savings getDSTStart returns correct start date United Kingdom 2022 1`] = `"2022-03-27"`;

exports[`daylight savings getDSTStart returns correct start date United Kingdom 2023 1`] = `"2023-03-26"`;

exports[`daylight savings getDSTStart returns correct start date United Kingdom 2024 1`] = `"2024-03-31"`;

exports[`daylight savings getDSTStart returns correct start date United States 2021 1`] = `"2021-03-14"`;

exports[`daylight savings getDSTStart returns correct start date United States 2022 1`] = `"2022-03-13"`;

exports[`daylight savings getDSTStart returns correct start date United States 2023 1`] = `"2023-03-12"`;

exports[`daylight savings getDSTStart returns correct start date United States 2024 1`] = `"2024-03-10"`;
